src/olympia/api/throttling.py

```diff
@@ -99,6 +100,38 @@ def allow_request(self, request, view):
             return True
 
 
+class CheckThrottlesFormMixin:
+    """
+    Mixin to add to a django form to check DRF throttles when running clean().
+
+    Requires self.request and self.throttle_classes to be set on the form
+    instance. self.throttled_error_message can be used to customize the error
+    message on throttled requests.
+    """
+
+    throttled_error_message = _(
+        'You have submitted too many uploads recently. '
+        'Please try again after some time.'
```

Comment:
nit: this is, imo, an odd default message for a re-usable class to be used for django forms - the message form the abuse form is more generic.

Reply:
Totally agree, cf68c6c54b944edc6e52f4a0ee63922919e502ea